security/pkg/registry/registry.go

```diff
@@ -0,0 +1,115 @@
+// Copyright 2017 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package registry
+
+import (
+	"sync"
+
+	"github.com/golang/glog"
+	"k8s.io/api/core/v1"
+
+	"istio.io/istio/pilot/platform/kube"
+)
+
+// Registry is the standard interface for identity registry
+// implementation
+type Registry interface {
+	Check(string, string) bool
+	AddMapping(string, string)
+	DeleteMapping(string, string)
+}
+
+// IdentityRegistry is a naive registry that maintains a mapping between
+// identities (as strings)
+type IdentityRegistry struct {
+	sync.RWMutex
+	Map map[string]string
+}
+
+// Check checks whether id1 is mapped to id2
+func (reg *IdentityRegistry) Check(id1, id2 string) bool {
+	reg.RLock()
+	mapped, ok := reg.Map[id1]
+	reg.RUnlock()
+	if !ok {
+		return false
```

Comment:
Can we log some info here for debugging purpose? And if possible, please also log the case that mapped existed, but not equal to id2.

Reply:
Will do.